### PR TITLE
Filter interface types updated

### PR DESCRIPTION
**Pull request by tenshouuu**
## Description

Incorrect Filter type. Value can be undefined, see:
  https://github.com/ChainSafe/web3.js/blob/06610f21c9b5a261ab342f1d9ef7c9df55677f95/packages/web3-eth-contract/src/index.js#L413

Issue: We cannot correctly use types with optional(undefined) fields with the current Filter.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.

### CODE REVIEW

**Review by koraykoska — APPROVED** [review]
Yeah you are right. Awaiting another review before merging.

**Comment by NguyenNam449** [comment]
G

**Pull request closed without merging**
